File: 2932-maximum-strong-pair-xor-i/2932-maximum-strong-pair-xor-i.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var maximumStrongPairXor = function(nums) {
    let max_xor = 0;
    const n = nums.length;
    
    for (let i = 0; i < n; i++) {
        for (let j = i; j < n; j++) {
            let x = nums[i], y = nums[j];
            if (Math.abs(x - y) <= Math.min(x, y)) {
                max_xor = Math.max(max_xor, x ^ y);
            }
        }
    }
    
    return max_xor;
};

File: 2932-maximum-strong-pair-xor-i/2932-maximum-strong-pair-xor-i.rs
impl Solution {
    pub fn maximum_strong_pair_xor(nums: Vec<i32>) -> i32 {
        let mut max_xor = 0;
        let n = nums.len();
        
        for i in 0..n {
            for j in i..n {
                let x = nums[i];
                let y = nums[j];
                if (x - y).abs() <= x.min(y) {
                    max_xor = max_xor.max(x ^ y);
                }
            }
        }
 
        max_xor
    }
}